### DISCUSSION TITLE
QUDI Installation - Python version

### DISCUSSION THREAD

**Discussion by AItah** [Ulm-IQO/qudi-core#77] [Q&A] [answered]
Hi,

Does QUDI works only with Python 3.9.x ?

According to the installation instructions:
"While this is to our knowledge the most robust and preferred way of setting up a Python environment for qudi, it has a small disadvantage. You can not change the Python version for the environment, meaning you can only set up an environment with the same Python version you created the environment with.
So make sure the Python interpreter you use for calling the following commands has version 3.9.x. If you are missing Python 3.9 on your system, you can download and install the right version from https://www.python.org/."

Can we use latest Python version 3.12.0?

BR,
Amir

**Comment by Neverhorst** [comment, marked as answer]
The current version of `qudi` works with minimal Python version `3.8.x` and maximum version `3.10.x`. I will update the documentation accordingly. Thank you for pointing that out.
The reason why `qudi` does not support versions higher than `3.10` is the dependency on `PySide2` which does not allow for more modern Python versions. Before we can support higher Python versions, we need to migrate `qudi` to `PySide6` first.

**Reply by AItah** [reply]
10x!
(indeed, I thought that I saw something about it inside setup.py )
